pkgs/by-name/un/unblob/package.nix

```diff
@@ -0,0 +1,129 @@
+{
+  lib,
+  python3,
+  fetchFromGitHub,
+  gitUpdater,
+  makeWrapper,
+  e2fsprogs,
+  jefferson,
+  lz4,
+  lziprecover,
+  lzop,
+  p7zip,
+  sasquatch,
+  sasquatch-v4be,
+  simg2img,
+  ubi_reader,
+  unar,
+  zstd,
+}:
+
+let
+  # These dependencies are only added to PATH
+  runtimeDeps = [
+    e2fsprogs
+    jefferson
+    lziprecover
+    lzop
+    p7zip
+    sasquatch
+    sasquatch-v4be
+    ubi_reader
+    simg2img
+    unar
+    zstd
+    lz4
+  ];
+in
+python3.pkgs.buildPythonApplication rec {
+  pname = "unblob";
+  version = "24.12.4";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "onekey-sec";
+    repo = "unblob";
+    tag = version;
+    hash = "sha256-lQiY5CDz4y2cOGh4gYjRPUC7jLbbvredEBO/vPQN5Eo=";
+    forceFetchGit = true;
+    fetchLFS = true;
+  };
+
+  strictDeps = true;
+
+  build-system = with python3.pkgs; [ poetry-core ];
+
+  dependencies = with python3.pkgs; [
+    arpy
+    attrs
+    click
+    cryptography
+    dissect-cstruct
+    lark
+    lief.py
+    python3.pkgs.lz4 # shadowed by pkgs.lz4
+    plotext
+    pluggy
+    pyfatfs
+    pyperscan
+    python-magic
+    rarfile
+    rich
+    structlog
+    treelib
+    unblob-native
+  ];
+
+  nativeBuildInputs = [
+    makeWrapper
+  ];
+
+  # These are runtime-only CLI dependencies, which are used through
+  # their CLI interface
+  pythonRemoveDeps = [
+    "jefferson"
+    "ubi-reader"
+  ];
+
+  pythonImportsCheck = [ "unblob" ];
+
+  makeWrapperArgs = [
+    "--prefix PATH : ${lib.makeBinPath runtimeDeps}"
+  ];
+
+  nativeCheckInputs =
+    with python3.pkgs;
+    [
+      pytestCheckHook
+      pytest-cov
```

Comment:
This could be pytest-cov-stub

Reply:
Can you expand on that? Didn't work [before](https://github.com/NixOS/nixpkgs/pull/345554#discussion_r1795188207) nor does it work now `ModuleNotFoundError: No module named 'pytest_cov.embed'`

Reply:
(background: tests need to setup coverage to work in subprocesses with signals, see https://pytest-cov.readthedocs.io/en/latest/subprocess-support.html#signal-handlers)